### PR TITLE
docker trigger - remove tag dropdown from definition and make it into a input that takes in a regex

### PR DESCRIPTION
**Pull request by tomaslin**

### CODE REVIEW

**Comment by tomaslin** [comment]
@lwander PTAL

**Comment by tomaslin** [comment]
tested the config without tag and triggering still works with latest echo

**Comment by lwander** [comment]
Does it still work with regex?

**Comment by tomaslin** [comment]
@lwander tag is now in as an input box that will default to null if the string is empty. Seems to fix the triggering.

File: app/scripts/modules/core/pipeline/config/triggers/docker/dockerTrigger.module.js [hunk 1]
```diff
@@ -53,32 +52,15 @@ module.exports = angular.module('spinnaker.core.pipeline.trigger.docker', [
...
         trigger.repository = null;
       }
-      updateTagList();
+      updateTag();
     }
 
-    function updateTagList() {
-      if (!$scope.repositoryMap) {
-        return;
+    function updateTag(){
```
Comment (lwander) [inline app/scripts/modules/core/pipeline/config/triggers/docker/dockerTrigger.module.js]: Missing space before `{`

File: app/scripts/modules/core/pipeline/config/triggers/docker/dockerTrigger.module.js [hunk 2]
```diff
@@ -53,32 +52,15 @@ module.exports = angular.module('spinnaker.core.pipeline.trigger.docker', [
...
+      if(trigger.tag !== null && trigger.tag.trim() === ''){
```
Comment (lwander) [inline app/scripts/modules/core/pipeline/config/triggers/docker/dockerTrigger.module.js]: See above comment

**Comment by lwander** [comment]
LGTM, verified locally too.

**Pull request merged**